cmd/esc/cli/env_get.go

```diff
@@ -155,7 +153,86 @@ func (get *envGetCommand) showValue(
 	if err != nil {
 		return fmt.Errorf("getting environment: %w", err)
 	}
-	return get.env.renderValue(get.env.esc.stdout, env, path, format, true, showSecrets)
+	return get.env.renderValue(out, env, path, format, true, showSecrets)
+}
+
+func (get *envGetCommand) showValue(
+	ctx context.Context,
+	orgName string,
+	envName string,
+	revisionOrTag string,
+	path resource.PropertyPath,
+	format string,
+	showSecrets bool,
+) error {
+	return get.writeValue(ctx, get.env.esc.stdout, orgName, envName, revisionOrTag, path, format, showSecrets)
+}
+
+func diff(oldName, old, newName, new string) string {
+	edits := myers.ComputeEdits(span.URIFromPath(oldName), old, new)
+	return fmt.Sprint(gotextdiff.ToUnified(oldName, newName, old, edits))
+}
+
+func (get *envGetCommand) diff(
+	baseRevisionOrTag string,
+	baseEnv *envGetTemplateData,
+	diffRevisionOrTag string,
+	diffEnv *envGetTemplateData,
+) envDiffTemplateData {
+	valueDiff := diff(baseRevisionOrTag, baseEnv.Value, diffRevisionOrTag, diffEnv.Value)
+	defDiff := diff(baseRevisionOrTag, baseEnv.Definition, diffRevisionOrTag, diffEnv.Definition)
+	return envDiffTemplateData{Value: valueDiff, Definition: defDiff}
+}
+
+func (get *envGetCommand) diffValue(
+	ctx context.Context,
+	orgName string,
+	envName string,
+	baseRevisionOrTag string,
+	tipRevisionOrTag string,
+	path resource.PropertyPath,
+	format string,
+	showSecrets bool,
+) error {
+	var base strings.Builder
+	if err := get.writeValue(ctx, &base, orgName, envName, baseRevisionOrTag, path, format, showSecrets); err != nil {
+		return err
+	}
+	var tip strings.Builder
+	if err := get.writeValue(ctx, &tip, orgName, envName, tipRevisionOrTag, path, format, showSecrets); err != nil {
+		return err
+	}
+
+	baseRef := fmt.Sprintf("%s:%s", envName, baseRevisionOrTag)
```

Comment:
Did we land on @ instead? Assume that would be a separate PR

Reply:
We did--that'll be a separate PR